### PR TITLE
Filter for the standardizing the data as well as to get the full resu…

### PR DESCRIPTION
**Pull request by ghassanmas**
…lts from the Guardian api

Relates #25

### CODE REVIEW

**Comment by harrygfox** [comment]
Awaiting internal review from @nervnews/developers 
Making great progress, Team!

File: src/api/guardian.js [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
   const welvonAPI = new Guardian(process.env.GUARDIAN_KEY, false);
   welvonAPI.content
     .search(query, {
-      "show-fields": "headline,trailText,thumbnail,bodyText,page,lastModified"
+      "show-fields": "headline,trailText,thumbnail,bodyText,page,lastModified",
+      page: page
     })
     .then(response => {
-      cb(JSON.parse(response.body.response));
+      json_res = JSON.parse(response.body).response;
```
Comment (Karyum) [inline src/api/guardian.js]: what is json_res ? what kind of variable ?

File: src/api/guardian.js [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
+      const { currentPage, pageSize } = json_res;
```
Comment (Karyum) [inline src/api/guardian.js]: why not just do `const { currentPage, pageSize } = JSON.parse(response.body)`

**Comment by harrygfox** [comment]
@ghassanmas merge conflicts to resolve!

File: src/api/guardian.js [hunk 3]
```diff
@@ -1,16 +1,24 @@
...
+      if (currentPage == 1) allArticles = [];
+      if (currentPage + 1 <= pageSize && currentPage < 5) {
+        allArticles = allArticles.concat(guardian_filter(json_res));
+        guardianListing(query, currentPage + 1, cb);
```
Comment (Karyum) [inline src/api/guardian.js]: i love this

File: src/api/guardian_filter.js [hunk 4]
```diff
@@ -1 +1,13 @@
-module.exports = response => {};
+module.exports = response => {
+  let results = response.results.reduce((articles, article) => {
```
Comment (Karyum) [inline src/api/guardian_filter.js]: any reason why this is let and not a const ?

Comment (Karyum) [reply src/api/guardian_filter.js]: also you could use deconstruction for `(articles, article)` to be `(articles, { fields, id })`   so if you scroll down a bit to my next comment :arrow_down:

File: src/api/guardian_filter.js [hunk 5]
```diff
@@ -1 +1,13 @@
...
+    articles.push({
+      id: article.id,
```
Comment (Karyum) [inline src/api/guardian_filter.js]: instead of `id: article.id`  you could do just `id` now

File: src/api/guardian_filter.js [hunk 6]
```diff
@@ -1 +1,13 @@
...
+      image: article.fields.thumbnail,
+      headline: article.fields.headline,
+      trailText: article.fields.trailText,
+      bodyText: article.fields.bodyText
```
Comment (Karyum) [inline src/api/guardian_filter.js]: also for those now you just do `fields.x` ( x is whatever )

File: src/router.js [hunk 7]
```diff
@@ -1,20 +1,19 @@
...
 router.post("/articles", (req, res) => {
-query = req.body.search;
-    guardianListing(query, (results) => {
-      const articles = results.response.results
-        res.status(200).render('articles', { articles });
-      })
-})
+  query = req.body.search;
+  guardianListing(query, 1, results => {
+    const articles = results;
```
Comment (Karyum) [inline src/router.js]: why do you need this variable just pass the results to res.render

Comment (sophieeejw) [reply src/router.js]: yes, it has to have the same name when we pass it

Comment (ghassanmas) [reply src/router.js]: because it's called articles in the handlebars view, it has to match

File: src/api/guardian_filter.js (hunk 4 above)
Comment (ghassanmas) [reply src/api/guardian_filter.js]: done

File: src/api/guardian_filter.js (hunk 6 above)
Comment (ghassanmas) [reply src/api/guardian_filter.js]: done

File: src/api/guardian_filter.js (hunk 5 above)
Comment (ghassanmas) [reply src/api/guardian_filter.js]: done

File: src/api/guardian.js (hunk 3 above)
Comment (ghassanmas) [reply src/api/guardian.js]: it loves you too

File: src/api/guardian.js (hunk 2 above)
Comment (ghassanmas) [reply src/api/guardian.js]: I need to use json_res as well

File: src/api/guardian.js (hunk 1 above)
Comment (ghassanmas) [reply src/api/guardian.js]: the JSON reponse of guardian API

Comment (claireinez) [reply src/api/guardian.js]: Also why is it snake_cased when all your other variables are camelCased?

File: src/api/guardian.js [hunk 8]
```diff
@@ -1,14 +1,24 @@
...
   const welvonAPI = new Guardian(process.env.GUARDIAN_KEY, false);
   welvonAPI.content
     .search(query, {
-      "show-fields": "headline,trailText,thumbnail,bodyText,page,lastModified"
+      "show-fields": "headline,trailText,thumbnail,bodyText,page,lastModified",
+      page: page
     })
     .then(response => {
-      cb(JSON.parse(response.body));
+      json_res = JSON.parse(response.body).response;
```
Comment (Karyum) [inline src/api/guardian.js]: why is it a snake_case while every other varialbe is camel case?

**Pull request merged**